internal/engine/live_update_build_and_deployer.go

```diff
@@ -141,7 +142,10 @@ func (lubad *LiveUpdateBuildAndDeployer) buildAndDeploy(ctx context.Context, cu
 			_ = pw.Close()
 		}
 	}()
-	var archiveReader io.Reader = pr
+	tarBytes, err := ioutil.ReadAll(pr)
```

Comment:
I think this is going to explode memory usage on large files. Try it out with a large, multi-GB file (`dd if=/dev/zero of=file.txt count=1024 bs=1048576`) and running `tilt up`. Then create a memory profile by pressing Ctrl+O in the terminal UI.

You can check out the profile with pprof by running `go tool pprof tilt.heap_profile`.

Reply:
That makes sense, and I don't have the internet bandwidth to test this right now asdfgh (like I literally can't even pull the `golang` docker image) so I pushed up a version where we recreate the tar archive every time, and if tests pass i'll land -- can verify performance later.